velox/type/Type.h

```diff
@@ -1121,6 +1122,58 @@ FOLLY_ALWAYS_INLINE bool Type::isIntervalDayTime() const {
   return (this == INTERVAL_DAY_TIME().get());
 }
 
+constexpr long kMonthInYear = 12;
+/// Time interval in months.
+class IntervalYearMonthType : public IntegerType {
+ private:
+  IntervalYearMonthType() = default;
+
+ public:
+  static const std::shared_ptr<const IntervalYearMonthType>& get() {
+    static const std::shared_ptr<const IntervalYearMonthType> kType{
+        new IntervalYearMonthType()};
+    return kType;
+  }
+
+  const char* name() const override {
+    return "INTERVAL YEAR TO MONTH";
+  }
+
+  bool equivalent(const Type& other) const override {
+    // Pointer comparison works since this type is a singleton.
+    return this == &other;
+  }
+
+  std::string toString() const override {
+    return name();
+  }
+
+  /// Returns the interval 'value' (months) formatted as YEARS MONTHS.
+  /// For example, (INTERVAL '1-2' YEAR TO MONTH) would be represented as 1 2
```

Comment:
Please also indicate in the comment what would be the value for INTERVAL '1-2' YEAR TO MONTH. From the logic seems like that would be 14 months.

"For example, value 14 months is (INTERVAL '1-2' YEAR TO MONTH) would be represented as 1 2..."